app/Utils/HasPeriodicEvent.php

```diff
@@ -96,6 +97,10 @@ final public function updatePeriodicEvent(Semester $semester, Carbon $start_date
             }
             $event->save();
             $event->refresh();
+
+            Cache::put($this->underlyingControllerName . '_start_date', $start_date);
```

Comment:
I see why do you want caching here, but I would rather make it in a separate issue or in https://github.com/EotvosCollegium/mars/pull/555 as it will also rename underlyingControllerName.